src/java/org/apache/cassandra/service/paxos/PrepareVerbHandler.java

```diff
@@ -45,12 +46,21 @@ public void doVerb(Message<Commit> message)
         // Prepare phase incorporates a read to check the cas condition, so a read sensor is registered in addition to the write sensor
         sensors.registerSensor(context, Type.READ_BYTES);
         sensors.registerSensor(context, Type.WRITE_BYTES);
+        sensors.registerSensor(context, Type.INTERNODE_BYTES);
+        sensors.incrementSensor(context, Type.INTERNODE_BYTES, message.payloadSize(MessagingService.current_version));
         ExecutorLocals locals = ExecutorLocals.create(sensors);
         ExecutorLocals.set(locals);
 
         Message.Builder<PrepareResponse> reply = message.responseWithBuilder(doPrepare(message.payload));
+
         SensorsCustomParams.addWriteSensorToResponse(reply, sensors, context);
         SensorsCustomParams.addReadSensorToResponse(reply, sensors, context);
+
+        // calculate outbound internode bytes before adding the sensor to the response
+        int size = reply.currentPayloadSize(MessagingService.current_version);
```

Comment:
current_version is not accurate, the message could have been sent using a different version. Probably that's not that important.

Reply:
Good point @jtgrabowski and I believe it is important to use the right version when calculating the size b/c otherwise we might mess up while reading the data (fields and their sizes can change over different version...). 
The message version seems to be in the connection state (I'm a bit surprised that message version is not in the header)

Reply:
Looking deeper into C* code I've found this:
```
  /*
     * Size used for capacity enforcement purposes. Using current messaging version no matter what the peer's version is.
     */
    private int canonicalSize(Message<?> message)
    {
        return message.serializedSize(current_version);
    }
```
The value returned might not be fully correct (if established version is different), however maybe it's good enough for an estimate... So maybe we just keep it as it is.

Reply:
Good points. I think for UCU's the estimates are good enough. Also doesn't see there is easy access to the actual version in the verb handlers.